src/View/Cell.php

```diff
@@ -160,14 +179,31 @@ public function render($template = null)
         if ($template === null) {
             $template = $this->template;
         }
-        $builder = $this->viewBuilder();
-        $builder->layout(false);
-        $builder->template($template);
 
         $cache = [];
         if ($this->_cache) {
-            $cache = $this->_cacheConfig($template);
+            $cache = $this->_cacheConfig($this->action);
+            $result = Cache::read($cache['key'], $cache['config']);
+            if ($result !== false) {
+                return $result;
+            }
+        }
+
+        try {
+            $reflect = new ReflectionMethod($this, $this->action);
+            $reflect->invokeArgs($this, $this->args);
+        } catch (ReflectionException $e) {
+            throw new BadMethodCallException(sprintf(
+                'Class %s does not have a "%s" method.',
+                get_class($this),
+                $this->action
+            ));
         }
+
+        $builder = $this->viewBuilder();
+        $builder->layout(false);
+        $builder->template($template);
```

Comment:
Nope, I tried but it doesn't work because in that case the View instance gets created too early and `viewVars` and other view options don't get passed to the view since the action has still not been invoked.

I can get rid of View::cache() usage and just write to cache key since check for existing cached value is already done earlier.